### PR TITLE
MB-1384 Transit times parser

### PR DESCRIPTION
**Pull request by donaldthai**
## Description

Shamelessly forked code from the pricing parser.

The GHC transit time parser to parse domestic transit times. The parser helped to produced [this migration](https://github.com/transcom/mymove/pull/3570) which produces a csv file so we can import into the database.

## Reviewer Notes

- Work in progress, needs some polishing

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

1. Build parser binary
```sh
make bin/ghc-transit-time-parser
```

2. Download [XSLX here](https://docs.google.com/spreadsheets/d/1p2L_rvf-hsqp8djsSamuF1Dby5QxGSKS/edit#gid=1617474798) 

3. Run parser to generate a CSV file to import into your database table
```sh
bin/ghc-transit-time-parser --display --save-csv --filename [path to your file]/Appendix_C\(i\)_-_Transit_Time_Tables.xlsx 
```

4. Use your favorite database tool to import the data (ie. TablePlus)

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.

* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-1384) for this change

### CODE REVIEW

File: cmd/ghc-transit-times-parser/main.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	// Create a connection to the DB
+	db, err := cli.InitDatabase(v, nil, logger)
+	if err != nil {
+		// No connection object means that the configuraton failed to validate and we should not startup
+		// A valid connection object that still has an error indicates that the DB is not up and we should not startup
+		logger.Fatal("Connecting to DB", zap.Error(err))
+	}
+	defer func() {
```
Comment (Ryan-Koch) [inline cmd/ghc-transit-times-parser/main.go]: Nice use of `defer`

Comment (donaldthai) [reply cmd/ghc-transit-times-parser/main.go]: Shamelessly taken from the pricing parser!

File: pkg/parser/transittimes/parse.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+		} else {
+			log.Printf("No verify function for sheet index %d with Description %s\n", sheetIndex, description)
+		}
+	} else {
+		log.Print("Skip running the verify functions")
+	}
+
+	// Call process function
+	if len(xlsxInfo.ProcessMethods) > 0 {
+		for methodIndex, p := range xlsxInfo.ProcessMethods {
```
Comment (Ryan-Koch) [inline pkg/parser/transittimes/parse.go]: Is there a more verbose name we can use for this `p` variable?

File: pkg/parser/transittimes/parse.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+// Parse parses the XLSX file
+func Parse(xlsxDataSheets []XlsxDataSheetInfo, params ParamConfig, db *pop.Connection, logger Logger) error {
+	// Must be after processing config param
+	// Run the process function
+
+	err := db.Transaction(func(tx *pop.Connection) error {
+		tableFromSliceCreator := dbtools.NewTableFromSliceCreator(tx, logger, params.UseTempTables, params.DropIfExists)
+
+		if params.ProcessAll == true {
```
Comment (Ryan-Koch) [inline pkg/parser/transittimes/parse.go]: Could get away with `if params.ProcessAll {...}` though I do appreciate the reach for explicit-ness.

File: pkg/parser/transittime/parse.go [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+	return xlsxDataSheets
+}
+
+// Parse parses the XLSX file
+func Parse(xlsxDataSheets []XlsxDataSheetInfo, params ParamConfig, db *pop.Connection, logger Logger) error {
+	// Must be after processing config param
+	// Run the process function
+
+	err := db.Transaction(func(tx *pop.Connection) error {
+		tableFromSliceCreator := dbtools.NewTableFromSliceCreator(tx, logger, params.UseTempTables, params.DropIfExists)
```
Comment (garrettqmartin8) [inline pkg/parser/transittime/parse.go]: First time I've seen `dbtools.NewTableFromSliceCreator`. Pretty cool.

**Pull request merged**